Description: Fix build for machines that have 48-bit virtual addresses
Author: Edmund Grimley Evans <edmund.grimley.evans@gmail.com>
Bug: http://xcas.e.ujf-grenoble.fr/XCAS/viewtopic.php?f=4&t=1785

--- a/src/Equation.cc
+++ b/src/Equation.cc
@@ -616,7 +616,7 @@
 	return Equation_compute_size(arg,a,windowhsize,contextptr);
       gen tmp;
 #ifdef SMARTPTR64
-      * ((longlong * ) &tmp) = longlong(new ref_fraction(Tfraction<gen>(arg._VECTptr->front(),arg._VECTptr->back()))) << 16;
+      * ((ulonglong * ) &tmp) = ulonglong(new ref_fraction(Tfraction<gen>(arg._VECTptr->front(),arg._VECTptr->back()))) << 16;
 #else
       tmp.__FRACptr = new ref_fraction(Tfraction<gen>(arg._VECTptr->front(),arg._VECTptr->back()));
 #endif
--- a/src/sym2poly.cc
+++ b/src/sym2poly.cc
@@ -406,8 +406,8 @@
 #ifndef SMARTPTR64
 	  n.__POLYptr->ref_count==1 && d.__POLYptr->ref_count==1
 #else
-	  ((ref_polynome*)(* (longlong *) &n >> 16))->ref_count ==1 &&
-	  ((ref_polynome*)(* (longlong *) &d >> 16))->ref_count ==1	  
+	  ((ref_polynome*)(* (ulonglong *) &n >> 16))->ref_count ==1 &&
+	  ((ref_polynome*)(* (ulonglong *) &d >> 16))->ref_count ==1	  
 #endif
 	  ){
 	simplify(*n._POLYptr,*d._POLYptr,pptr->t);
--- a/src/usual.cc
+++ b/src/usual.cc
@@ -9866,14 +9866,14 @@
   const gen & grad2rad_g=grad2rad_g_;
 #else
     // Warning this does not work on ia64 with -O2
-  const define_alias_gen(alias_rad2deg_g,_DOUBLE_, (*(longlong *)&rad2deg_d) >> 8,(*(longlong *)&rad2deg_d)>>16);
+  const define_alias_gen(alias_rad2deg_g,_DOUBLE_, (*(ulonglong *)&rad2deg_d) >> 8,(*(ulonglong *)&rad2deg_d)>>16);
   const gen & rad2deg_g = *(const gen*) & alias_rad2deg_g;
-  const define_alias_gen(alias_deg2rad_g,_DOUBLE_, (*(longlong *)&deg2rad_d) >> 8,(*(longlong *)&deg2rad_d)>>16);
+  const define_alias_gen(alias_deg2rad_g,_DOUBLE_, (*(ulonglong *)&deg2rad_d) >> 8,(*(ulonglong *)&deg2rad_d)>>16);
   const gen & deg2rad_g = *(const gen*) & alias_deg2rad_g;
   //grad
-  const define_alias_gen(alias_rad2grad_g,_DOUBLE_, (*(longlong *)&rad2grad_d) >> 8,(*(longlong *)&rad2grad_d)>>16);
+  const define_alias_gen(alias_rad2grad_g,_DOUBLE_, (*(ulonglong *)&rad2grad_d) >> 8,(*(ulonglong *)&rad2grad_d)>>16);
   const gen & rad2grad_g = *(const gen*) & alias_rad2grad_g;
-  const define_alias_gen(alias_grad2rad_g, _DOUBLE_, (*(longlong *)&grad2rad_d) >> 8, (*(longlong *)&grad2rad_d) >> 16);
+  const define_alias_gen(alias_grad2rad_g, _DOUBLE_, (*(ulonglong *)&grad2rad_d) >> 8, (*(ulonglong *)&grad2rad_d) >> 16);
   const gen & grad2rad_g = *(const gen*)& alias_grad2rad_g;
 #endif
 
